specification/compute/resource-manager/Microsoft.Compute/stable/2020-09-30/examples/UpdateAManagedDiskToAddPurchasePlan.json

```diff
@@ -0,0 +1,80 @@
+{
+  "parameters": {
+    "subscriptionId": "{subscription-id}",
+    "resourceGroupName": "myResourceGroup",
+    "api-version": "2020-09-30",
+    "diskName": "myDisk",
+    "disk": {
+      "properties": {
+        "purchasePlan": {
+          "name": "myPurchasePlanName",
+          "publisher": "myPurchasePlanPublisher",
+          "product": "myPurchasePlanProduct",
+          "promotionCode": "myPurchasePlanPromotionCode"
+        }
+      }
+    }
+  },
+  "responses": {
+    "202": {
+      "body": {
+        "name": "myDisk",
+        "location": "westus",
+        "sku": {
+          "name": "Standard_LRS",
+          "tier": "Standard"
+        },
+        "properties": {
+          "osType": "Windows",
+          "hyperVGeneration": "V1",
+          "purchasePlan": {
+            "name": "myPurchasePlanName",
+            "publisher": "myPurchasePlanPublisher",
+            "product": "myPurchasePlanProduct",
+            "promotionCode": "myPurchasePlanPromotionCode"
+          },
+          "creationData": {
+            "createOption": "FromImage",
+            "imageReference": {
+              "id": "/Subscriptions/{subscription-id}/Providers/Microsoft.Compute/Locations/westus/Publishers/test_test_pmc2pc1/ArtifactTypes/VMImage/Offers/marketplace_vm_test/Skus/test_sku/Versions/1.0.0"
+            }
+          },
+          "diskSizeGB": 127,
+          "provisioningState": "Updating",
+          "isArmResource": true,
+          "faultDomain": 0
+        }
+      }
+    },
+    "200": {
+      "body": {
+        "name": "myDisk",
+        "location": "westus",
+        "sku": {
+          "name": "Standard_LRS",
+          "tier": "Standard"
+        },
+        "properties": {
+          "osType": "Windows",
+          "hyperVGeneration": "V1",
+          "purchasePlan": {
+            "name": "myPurchasePlanName",
+            "publisher": "myPurchasePlanPublisher",
+            "product": "myPurchasePlanProduct",
+            "promotionCode": "myPurchasePlanPromotionCode"
+          },
+          "creationData": {
+            "createOption": "FromImage",
+            "imageReference": {
+              "id": "/Subscriptions/{subscription-id}/Providers/Microsoft.Compute/Locations/westus/Publishers/test_test_pmc2pc1/ArtifactTypes/VMImage/Offers/marketplace_vm_test/Skus/test_sku/Versions/1.0.0"
+            }
+          },
+          "diskSizeGB": 127,
+          "provisioningState": "Succeeded",
+          "isArmResource": true,
+          "faultDomain": 0
```

Comment:
isArmResource and faultDomain are internal only properties and should not appear in the response